webapi/src/test/scala/org/knora/webapi/slice/resourceinfo/repo/TestResourceInfoRepo.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.resourceinfo.repo
+
+import zio.Ref
+import zio.UIO
+import zio.ULayer
+import zio.ZIO
+import zio.ZLayer
+
+import org.knora.webapi.slice.resourceinfo.domain
+import org.knora.webapi.slice.resourceinfo.domain.InternalIri
+import org.knora.webapi.slice.resourceinfo.domain.ResourceInfo
+import org.knora.webapi.slice.resourceinfo.domain.ResourceInfoRepo
+
+final case class TestResourceInfoRepo(entitiesRef: Ref[Map[(InternalIri, InternalIri), List[ResourceInfo]]])
```

Comment:
I'm not sure if I understand why we need a repo for the `ResourceInfo`s. Is this used like a cache?

Reply:
No this is not caching, though in future this might be the integration point with caching .
A `Repository` is the data access object and hides the implementation details of accessing and mapping the data from the database from the rest of the application and exposes the domain object. Hence the `trait` lives in the domain package but the implementation is in another package as it contains no domain relevant logic.

Reply:
But shouldn't it then be a proper resource repo instead of a repo for the `ResourceInfo`s? `ResourceInfo` is not (or was at least not until now) a domain object. IMO we should be careful with defining new domain objects and be clear about the purpose of each one. Or do you think each route can have its own "domain objects" and it doesn't matter if there are several similar ones? I'm lacking the experience to see all the consequences of this.

Reply:
> But shouldn't it then be a proper resource repo instead of a repo for the ResourceInfos?

For the information which needs to be provided in this endpoint the `ResourceInfo` and its route/api representation `ResourceInfoDto` is sufficient. It became its domain object by the fact that we need an endpoint which is providing a small set of meta information like creation, lastModfication and deletion dates. For the lack of a better naming I  have called it `ResourceInfo`.

> Or do you think each route can have its own "domain objects" and it doesn't matter if there are several similar ones?
Yes and no.  Not every route will have its own domain object, most probably they will share domain objects and representations were it makes sense. For example the `ResourceInfo` might become a part of a bigger `Resource` domain object or aggregate in the future.